src/thunderbots/software/geom/polygon.cpp

```diff
@@ -0,0 +1,89 @@
+#include "geom/polygon.h"
+
+#include <unordered_set>
+
+#include "geom/util.h"
+
+// CMake defines NDEBUG for Release build configurations
+#ifndef NDEBUG
+#define VALIDATE_POLYGON
+#endif
+
+Polygon::Polygon(const std::vector<Segment>& _segments) : segments(_segments)
```

Comment:
Could we instead accept a `std::vector<Point>` here, and say that the polygon is formed by drawing lines between consecutive points, and from the last point to the first? Then we don't need to worry about checking anything, and it significantly reduces the probability that this will be called incorrectly?

Then we can also remove all the preoprocessor stuff?

Reply:
Constructor takes a vector of Points now, removed preprocessor stuff